fix: build new JsonLDObject of JsonLDObject with non-empty DEFAULT_JS…

rkxx opened the pull request:
The changes in the pom file are required to run the unit tests (kotlin). Feel free to only merge the source changes and keep the project pure java.

Without the correction the test ``createDerivedJsonLDObjectWithDefaultContexts`` fails, because the property ``contexts`` of the resulting ``jsonLDObject`` contains a listOf(URI, listOf(URI)) instead of a flat listOf(URI, URI)

peacekeeper reviewed (changes requested):
Hello thanks for the PR! Two comments:

1. I would indeed prefer to keep this pure Java rather than introducing Kotlin (especially if it's just for unit tests).
2. In JsonLDObject this PR changes a lot of the whitespace, newlines, etc. As a consequence, it's really hard to see what "actual" change this PR is making.

Could you update the PR to leave all the whitespace and formatting as-is, and only apply the actual change you're proposing?

rkxx commented:
Sorry, my fault. Intellij automatically ignored all formating changes so I wasn't aware of the mess. I have removed all formatting changes and unit tests. Is there any reason that you don't accept kotlin. It's really hard for me to fall back to plain Java. I did it for the existing sources, but would prefer to add new separated code using kotlin. Nevertheless, for the time being I removed the unit tests.

peacekeeper commented:
Thanks, please check if you are a member of DIF and the Identifiers&Discovery Working Group, so we can accept the contribution: https://identity.foundation/join/

peacekeeper commented:
Closing due to potential IPR issues.
